src/backend/tests/work-packages.test.ts

```diff
@@ -157,4 +159,34 @@ describe('Work Packages', () => {
   test('calculateWorkPackageProgress', async () => {
     expect(calculateWorkPackageProgress([], [])).toBe(0);
   });
+
+  test('getSingleWorkPackage fails if the work package does not exist', async () => {
+    jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(null);
+
+    await expect(
+      async () => await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 1 })
+    ).rejects.toThrow(new NotFoundException('Work Package', '1.1.1'));
+  });
+
+  test('getSingleWorkPackage fails if the project does not exist', async () => {
+    jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(null);
+
+    await expect(
+      async () => await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 100, workPackageNumber: 1 })
+    ).rejects.toThrow(new NotFoundException('Work Package', '1.100.1'));
+  });
+
+  test('getSingleWorkPackage fails if a project wbs number given (work_package number of 0)', async () => {
+    await expect(
+      async () => await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 0 })
+    ).rejects.toThrow(new HttpException(404, 'WBS Number 1.1.0 is a project WBS#, not a Work Package WBS#'));
+  });
+
+  // test('getSingleWorkPackage runs properly', async () => {
+  //   jest.spyOn(prisma.work_Package, 'findFirst').mockResolvedValue(prismaWorkPackage1);
+
+  //   const result = await WorkPackageService.getSingleWorkPackage({ carNumber: 1, projectNumber: 1, workPackageNumber: 1 });
```

Comment:
Wouldn't passing in 1.1.1 throw an error like your first test? You have to pass in the values that would return you prismaWorkPackage1